meteomatics/src/util.rs

```diff
@@ -0,0 +1,399 @@
+pub async fn df_add_latlon(df_in: polars::frame::DataFrame, point: &Point) -> 
+std::result::Result<polars::frame::DataFrame, polars::error::PolarsError> {
+    use polars::prelude::*;
+    // https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.shape
+    // Get (height, width) of the DataFrame. Get width:
+    let n = df_in.height();
+    // TODO: Convert to simplified version see postal code
+    let mut lat: Vec<f64> = Vec::new();
+    let mut lon: Vec<f64> = Vec::new();
+    for _ in 0..n {
+        lat.push(point.lat);
+        lon.push(point.lon);
+    }
+    // https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.extend
+    // https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.get_column_names
+    let df_tmp = df!("lat" => &lat, "lon" => &lon)?;
+    let df_out: DataFrame = df_tmp.hstack(df_in.get_columns())?;
+    Ok(df_out)
+}
+
+/// Creates a new DataFrame with added postal_Code extracted from the provided postal code.
+pub async fn df_add_postal(df_in: polars::frame::DataFrame, postal: &str) -> 
+std::result::Result<polars::frame::DataFrame, polars::error::PolarsError> {
+    // https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.shape
+    // Get (height, width) of the DataFrame. Get width:
+    let n = df_in.height();
+    let col_zipcode = vec![postal; n];
+    // https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.extend
+    // https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.get_column_names
+    let df_tmp = df!("station_id" => col_zipcode)?;
+    let df_out: DataFrame = df_tmp.hstack(df_in.get_columns())?;
+    Ok(df_out)
+}
+
+/// Convert the HTTP response body text for time series (i.e. the downloaded CSV) into a ```polars``` DataFrame. 
+/// Consumes the HTTP response.
+pub async fn parse_response_to_df(
+    response: Response,
+) -> std::result::Result<polars::frame::DataFrame, polars::error::PolarsError> {
+    // Get the response text:
+    let body = response.text().await.unwrap();
+
+    // Parse the response to a DataFrame
+    // TODO: rename variable
+    let file = std::io::Cursor::new(&body);
+    use polars::prelude::*; 
+    let df1 = polars::io::csv::CsvReader::new(file)
+        .infer_schema(Some(100))
+        .with_delimiter(b';')
+        .has_header(true)
+        .with_parse_dates(false)
+        .with_ignore_parser_errors(false)
+        .finish()?;
+
+    Ok(df1)
+}
+
+// Converts the HTTP response into a polars DataFrame.
+pub async fn parse_grid_response_to_df(
+    response: Response,
+) -> std::result::Result<polars::frame::DataFrame, polars::error::PolarsError> {
+        // Get the response text:
+        let body = response.text().await.unwrap();
+
+        // Parse the response to a DataFrame
+        // TODO: Rename variable
+        let file = std::io::Cursor::new(&body);
+        use polars::prelude::*; 
+        let df1 = polars::io::csv::CsvReader::new(file)
```

Comment:
df1 is too generic, maybe first_data_frame, or something more meaningful. It is hard to understand what it is this way.